lib/KalturaFTPServer.js

```diff
@@ -260,13 +297,13 @@ FtpSession.prototype = {
 			else {
 				switch (extension) {
 					case 'xml':
-						This.handleXmlListResponse(listResponse, callback, This, service, currentList, map, pager, currentClient, handler, filter);
+						This.handleXmlListResponse(listResponse, service, currentList, map, pagingHandler);
```

Comment:
Instead of passing currentList into the methods, append the results to currentList inside pagingHandler.
And instead of passing service into the handlers, call it before the handler and implement it only once for all three handlers.